snappy-core/src/main/scala/org/apache/spark/sql/SnappyContext.scala

```diff
@@ -347,29 +354,84 @@ class SnappyContext protected[spark](@transient override val sparkContext: Spark
-   *    val props = Map.empty[String, String]
-   *    val data = Seq(Seq(1, 2, 3), Seq(7, 8, 9), Seq(9, 2, 3), Seq(4, 2, 3), Seq(5, 6, 7))
-   *    val rdd = sc.parallelize(data, data.length).map(s => new Data(s(0), s(1), s(2)))
-   *    val dataDF = snc.createDataFrame(rdd)
-   *    snappyContext.createTable(tableName, "column", dataDF.schema, props)
-   *
-   * }}}
-   *
-   * @param tableName Name of the table
-   * @param provider Provider name such as 'COLUMN', 'ROW', 'JDBC', 'PARQUET' etc.
-   * @param schema   Table schema
-   * @param options  Properties for table creation. See options list for different tables.
-   *                 https://github.com/SnappyDataInc/snappydata/blob/master/docs/rowAndColumnTables.md
-   * @return DataFrame for the table
-   */
+    * Creates a Snappy managed table. Any relation providers (e.g. parquet, jdbc etc)
+    * supported by Spark & Snappy can be created here. Unlike SqlContext.createExternalTable this
+    * API creates a persistent catalog entry.
+    * {{{
+    *
+    * val airlineDF = snappyContext.createTable(stagingAirline, "parquet", Map("path" -> airlinefilePath))
+    *
+    * }}}
+    *
+    * @param tableName Name of the table
+    * @param provider  Provider name such as 'COLUMN', 'ROW', 'JDBC', 'PARQUET' etc.
+    * @param options   Properties for table creation
+    * @return DataFrame for the table
+    */
+  def createTable(
+      tableName: String,
+      provider: String,
+      options: java.util.Map[String, String]): DataFrame = {
+    createTable(tableName, provider, options.asScala.toMap)
+  }
+
+  /**
+    * Creates a Snappy managed table. Any relation providers (e.g. parquet, jdbc etc)
+    * supported by Spark & Snappy can be created here. Unlike SqlContext.createExternalTable this
+    * API creates a persistent catalog entry.
+    *
+    * {{{
+    *
+    *    case class Data(col1: Int, col2: Int, col3: Int)
+    *    val props = Map.empty[String, String]
+    *    val data = Seq(Seq(1, 2, 3), Seq(7, 8, 9), Seq(9, 2, 3), Seq(4, 2, 3), Seq(5, 6, 7))
+    *    val rdd = sc.parallelize(data, data.length).map(s => new Data(s(0), s(1), s(2)))
+    *    val dataDF = snc.createDataFrame(rdd)
+    *    snappyContext.createTable(tableName, "column", dataDF.schema, props)
+    *
+    * }}}
+    *
+    * @param tableName Name of the table
+    * @param provider  Provider name such as 'COLUMN', 'ROW', 'JDBC', 'PARQUET' etc.
+    * @param schema    Table schema
+    * @param options   Properties for table creation. See options list for different tables.
+    *                  https://github.com/SnappyDataInc/snappydata/blob/master/docs/rowAndColumnTables.md
+    * @return DataFrame for the table
+    */
+  def createTable(
+      tableName: String,
+      provider: String,
+      schema: StructType,
+      options: java.util.Map[String, String]): DataFrame = {
+    createTable(tableName, provider, schema, options.asScala.toMap);
+  }
+
```

Comment:
Yes. Those changes will be needed there as well. Rishi also mentioned some Optional class instead of Option,  that probably fit both for java and scala (need to check for python) . if that works out then we will not need to overload these methods.